test/sql/types/interval/test_interval.test

```diff
@@ -98,6 +98,12 @@ select interval '-05:12:34.567890' as test_interval;
 ----
 -05:12:34.56789
 
+# Issue #9631 - Parse hms > 24:00:00
+query I
+SELECT ('35:10:00'::interval);
```

Comment:
What about triple digit numbers/numbers that would overflow?

Reply:
Turns out that we can't overflow time because hours are 32 bit and time is 64 bit. So limiting to 10 digits solves both problems.